python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+            fwhmTot * float(np.sqrt(max(0.0, weightsHigh))),
+        )
+
+
+def _corrCovDiff(a: np.ndarray, b: np.ndarray) -> tuple[float, float, float]:
+    """
+    Compute Pearson correlation, covariance, and (a-b) variance for arrays.
+
+    Parameters
+    ----------
+    a : `ndarray`
+        First 1D array (component of rotated centroids).
+    b : `ndarray`
+        Second 1D array (component of rotated centroids).
+
+    Returns
+    -------
+    corr : `float`
+        Pearson correlation coefficient.
+    cov : `float`
+        Covariance scaled to centipix² (i.e., multiplied by ``100^2``).
+    dvar : `float`
+        Variance of (a - b) scaled to centipix² (i.e., multiplied by
+        ``100^2``).
+    """
+    if np.all(np.isnan(a)) or np.all(np.isnan(b)):
+        return np.nan, np.nan, np.nan
+    corr = float(np.corrcoef(a, b)[0, 1])
+    cov = float(np.cov(a, b, ddof=1)[0, 1]) * (100.0**2)
+    dvar = float(np.var(a - b, ddof=1)) * (100.0**2)
+    return corr, cov, dvar
+
+
+def _centers(df1: pd.DataFrame, df2: pd.DataFrame) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Compute mean (x, y) centers for two detector dataframes.
+
+    Parameters
+    ----------
+    df1 : `pandas.DataFrame`
+        First detector dataframe; must contain ``'xfp'``, ``'yfp'``.
+    df2 : `pandas.DataFrame`
+        Second detector dataframe; must contain ``'xfp'``, ``'yfp'``.
+
+    Returns
+    -------
+    center1 : `ndarray`
+        Shape (2,). Mean of (xfp, yfp) for ``df1``.
+    center2 : `ndarray`
+        Shape (2,). Mean of (xfp, yfp) for ``df2``.
+    """
+    c1 = np.array(
+        [
+            np.nanmean(pd.to_numeric(df1["xfp"], errors="coerce")),
+            np.nanmean(pd.to_numeric(df1["yfp"], errors="coerce")),
+        ],
+        float,
+    )
+    c2 = np.array(
+        [
+            np.nanmean(pd.to_numeric(df2["xfp"], errors="coerce")),
+            np.nanmean(pd.to_numeric(df2["yfp"], errors="coerce")),
+        ],
+        float,
+    )
+    return c1, c2
+
+
+def _alignTimes(df1: pd.DataFrame, df2: pd.DataFrame) -> tuple[np.ndarray | None, np.ndarray | None]:
+    """
+    Align two detector dataframes on common ``elapsed_time`` stamps.
+
+    Parameters
+    ----------
+    df1 : `pandas.DataFrame`
+        First detector dataframe; must contain ``'elapsed_time'``.
+    df2 : `pandas.DataFrame`
+        Second detector dataframe; must contain ``'elapsed_time'``.
+
+    Returns
+    -------
+    i1 : `ndarray` or `None`
+        Indices into ``df1`` for rows matching the common timestamps, or
+        ``None`` if no overlap.
+    i2 : `ndarray` or `None`
+        Indices into ``df2`` for rows matching the common timestamps, or
+        ``None`` if no overlap.
+    """
+    t, i1, i2 = np.intersect1d(df1["elapsed_time"], df2["elapsed_time"], return_indices=True)
+    return (i1, i2) if t.size else (None, None)
+
+
+def _correctedXy(df: pd.DataFrame) -> np.ndarray:
```

Comment:
Again, needs a verb in the name, so you could go with `getCorrectedXy()` perhaps.